manifests/aws_prune.pp

```diff
@@ -3,18 +3,30 @@
 # Sensu handler for removing clients from sensu if they are not in the AWS
 # API.
 #
-class sensu_handlers::aws_prune inherits sensu_handlers {
+class sensu_handlers::aws_prune (
+  $dependencies = {
+    'fog' => { provider => $gem_provider },
+    'unf' => { provider => $gem_provider },
+  },
+) inherits sensu_handlers {
 
-  # Only EC2 Sensu servers need to worry about querying the AWS api to know if 
+  # Only EC2 Sensu servers need to worry about querying the AWS api to know if
   # They need to prune or not
   if str2bool($::is_ec2) == true {
 
+    if $dependencies {
```

Comment:
I don't think you need the `if` block, if people don't want deps they can just use an empty hash.